src/modules/feature-limit/feature-limit.service.spec.ts

```diff
@@ -197,6 +197,169 @@ describe('FeatureLimitService', () => {
+      const userUuid = v4();
+      const tierId = v4();
+      const user = newUser({ attributes: { uuid: userUuid, tierId } });
+
+      const tierLimits = [
+        newFeatureLimit({
+          type: LimitTypes.Boolean,
+          label: LimitLabels.FileVersionEnabled,
+          value: 'true',
+        }),
+        newFeatureLimit({
+          type: LimitTypes.Counter,
+          label: LimitLabels.FileVersionMaxSize,
+          value: '10485760',
+        }),
+        newFeatureLimit({
+          type: LimitTypes.Counter,
+          label: LimitLabels.FileVersionRetentionDays,
+          value: '15',
+        }),
+        newFeatureLimit({
+          type: LimitTypes.Counter,
+          label: LimitLabels.FileVersionMaxNumber,
+          value: '10',
+        }),
+      ];
+
+      userRepository.findByUuid.mockResolvedValueOnce(user);
+      limitsRepository.findUserOverriddenLimitsByLabels.mockResolvedValueOnce(
+        [],
+      );
+      limitsRepository.findLimitsByLabelsAndTier.mockResolvedValueOnce(
+        tierLimits,
+      );
+
+      const result = await service.getFileVersioningLimits(userUuid);
+
+      expect(result).toEqual({
+        enabled: true,
+        maxFileSize: 10485760,
+        retentionDays: 15,
+        maxVersions: 10,
+      });
+    });
+
+    it('When user has overridden limits, then overrides should take precedence', async () => {
+      const userUuid = v4();
+      const tierId = v4();
+      const user = newUser({ attributes: { uuid: userUuid, tierId } });
+
+      const tierLimits = [
+        newFeatureLimit({
+          type: LimitTypes.Boolean,
+          label: LimitLabels.FileVersionEnabled,
+          value: 'true',
+        }),
+        newFeatureLimit({
+          type: LimitTypes.Counter,
+          label: LimitLabels.FileVersionMaxSize,
+          value: '10485760',
+        }),
+        newFeatureLimit({
+          type: LimitTypes.Counter,
+          label: LimitLabels.FileVersionRetentionDays,
+          value: '15',
+        }),
+        newFeatureLimit({
+          type: LimitTypes.Counter,
+          label: LimitLabels.FileVersionMaxNumber,
+          value: '10',
+        }),
+      ];
+
+      const overriddenLimits = [
+        newFeatureLimit({
+          type: LimitTypes.Counter,
+          label: LimitLabels.FileVersionMaxSize,
+          value: '20971520',
+        }),
+        newFeatureLimit({
+          type: LimitTypes.Counter,
+          label: LimitLabels.FileVersionMaxNumber,
+          value: '20',
+        }),
+      ];
+
+      userRepository.findByUuid.mockResolvedValueOnce(user);
+      limitsRepository.findUserOverriddenLimitsByLabels.mockResolvedValueOnce(
+        overriddenLimits,
+      );
+      limitsRepository.findLimitsByLabelsAndTier.mockResolvedValueOnce(
+        tierLimits,
+      );
+
+      const result = await service.getFileVersioningLimits(userUuid);
+
+      expect(result).toEqual({
+        enabled: true,
+        maxFileSize: 20971520,
```

Comment:
use overridenLimits[0].value, as it requires less maintenance and is far more semantic